### PR TITLE
Added new example for getting colormaps

### PR DESCRIPTION
**Pull request by ehsteve**
### Description
Provide a simple example to get at the SunPy colormaps and using them with matplotlib

### CODE REVIEW

**Comment by nabobalis** [comment]
So I rebased and decided that we also do not need the reference colormap example. I added it to the new example.

Since the CI is on fire, the only build we care about is the docs one.

**Comment by Cadair** [comment]
Can we set the second image (the one with all the colourmaps) as the thumbnail? i.e. https://sphinx-gallery.readthedocs.io/en/latest/auto_examples/plot_choose_thumbnail.html

**Comment by nabobalis** [comment]
And like magic.

File: examples/plotting/sunpy_matplotlib_colormap.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 This examples shows how you can use the SunPy colormaps with Matplotlib.
 Also the full range of colormaps we provide.
 """
+sphinx_gallery_thumbnail_number = 2
```
Comment (Cadair) [inline examples/plotting/sunpy_matplotlib_colormap.py]: this shouldn't be here.

File: examples/plotting/sunpy_matplotlib_colormap.py [hunk 2]
```diff
@@ -43,6 +44,7 @@
 print(cm.cmlist.keys())
 
 ###############################################################################
+# Ignore this: sphinx_gallery_thumbnail_number = 2
```
Comment (Cadair) [inline examples/plotting/sunpy_matplotlib_colormap.py]: or this.

File: examples/plotting/sunpy_matplotlib_colormap.py [hunk 3]
```diff
@@ -43,6 +44,7 @@
...
 # We also provide a function that will display all our colormaps:
```
Comment (Cadair) [inline examples/plotting/sunpy_matplotlib_colormap.py]: Add 
```
# sphinx_gallery_thumbnail_number = 2
```
here.

File: examples/plotting/sunpy_matplotlib_colormap.py (hunk 1 above)
Comment (nabobalis) [reply examples/plotting/sunpy_matplotlib_colormap.py]: GOD DAMN IT

File: examples/plotting/sunpy_matplotlib_colormap.py (hunk 3 above)
Comment (Cadair) [reply examples/plotting/sunpy_matplotlib_colormap.py]: as done here: https://sphinx-gallery.readthedocs.io/en/latest/_downloads/plot_choose_thumbnail.py

File: examples/plotting/sunpy_matplotlib_colormap.py (hunk 1 above)
Comment (nabobalis) [reply examples/plotting/sunpy_matplotlib_colormap.py]: I WILL FIX THIS

Comment (nabobalis) [reply examples/plotting/sunpy_matplotlib_colormap.py]: I HAVE THE POWER

**Pull request merged**